### PR TITLE
Document node interpolation (for non-leaf nodes)

### PR DESCRIPTION
**Pull request by odelalleau**
This is a follow-up to https://github.com/omry/omegaconf/commit/5f471ecbb6105ea5a1f7f4b422e7cd75499f0de8#r46668217

### CODE REVIEW

File: docs/source/usage.rst [hunk 1]
```diff
@@ -325,6 +325,27 @@ Example:
...
+    ... )
+    >>> cfg.player.height
+    180
+    >>> cfg.player.weight
+    75
+    >>> cfg.player = "${fred}"
+    >>> cfg.player.height
+    195
+    >>> cfg.player.weight
+    90
```
Comment (omry) [inline docs/source/usage.rst:347]: I think this simpler example is sufficient:

```suggestion
    >>> cfg = OmegaConf.create(
    ...     {
    ...         "john": {"height": 180, "weight": 75},
    ...         "player": "${john}",
    ...     }
    ... )
    >>> cfg.player
    {"height": 180, "weight": 75}
```

Comment (odelalleau) [reply docs/source/usage.rst]: Alright, I took your suggestion!

**Review by omry — APPROVED** [review]

**Comment by omry** [comment]
looks like the build failures are from the docs (wrong string quotes). can you address?

**Comment by odelalleau** [comment]
Oops, hopefully should be ok now

**Pull request merged**